dask/dataframe/core.py

```diff
@@ -5087,10 +5084,6 @@ def map_partitions(
             _meta=meta,
             **kwargs3
         )
-    elif not simple:
-        dsk = partitionwise_graph(
-            apply, name, func, *args2, **kwargs3, dependencies=dependencies
-        )
```

Comment:
Maybe `kwargs3` is more complex or unpleasant in some way?

Reply:
Oh I missed the kwargs / kwargs3 difference.

```python
(Pdb) pp kwargs
{'how': 'any',
 'subset': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k'],
 'thresh': None}
(Pdb) pp kwargs3
{'how': 'any',
 'subset': 'list-efe89017-6645-4acb-8d68-e29aa62f3040',
 'thresh': None}
```

So we want kwargs3 for sure. My latest commit uses kwargs3, but not `apply` in these cases. We'll see if CI passes.